pages/rating.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from "react";
+import {
+  Box,
+  Container,
+  Paper,
+  Typography,
+  Alert,
+  Grid,
+} from "@mui/material";
+import { GetServerSideProps, InferGetServerSidePropsType } from "next";
+import { ParsedUrlQuery } from "querystring";
+import { SendMail } from "../api/sendMail";
+import { styled } from "@mui/system";
+import Image from "next/image";
+import { ExpectedQuery, WrongQuery } from '../types/Rating';
+
+type Props = ExpectedQuery | WrongQuery;
+
+const isValidQuery = (query: ParsedUrlQuery): query is ExpectedQuery => {
+  return (
+    (query as ParsedUrlQuery).mail !== undefined &&
+    (query as ParsedUrlQuery).rating !== undefined
```

Comment:
Above type assertion is redundant, because since it doesn't change the type of expression(it inherits).